apps/server/src/modules/school/api/test/school-users.api.spec.ts

```diff
@@ -266,37 +268,215 @@ describe('School Controller (API)', () => {
 				return { loggedInClient };
 			};
 
-			it('should return 404', async () => {
+			it('should not return any users', async () => {
 				const { loggedInClient } = await setup();
 				const someId = new ObjectId().toHexString();
 
 				const response = await loggedInClient.get(`/${someId}/students`);
 
-				expect(response.status).toEqual(HttpStatus.NOT_FOUND);
+				const body = response.body as SchoolUserListResponse;
+
+				expect(response.status).toEqual(HttpStatus.OK);
+				expect(body.data.length).toEqual(0);
 			});
 		});
 
-		describe('when user has no permission to view students', () => {
+		describe('when user has no permission to view students but is teacher in class with users assigned', () => {
+			beforeEach(() => {
+				Configuration.set('TEACHER_STUDENT_VISIBILITY__IS_ENABLED_BY_DEFAULT', false);
+			});
+
+			afterEach(() => {
+				Configuration.set('TEACHER_STUDENT_VISIBILITY__IS_ENABLED_BY_DEFAULT', true);
+			});
+
 			const setup = async () => {
```

Comment:
Combine setup of this and the following tests (as they are nearly identical) and make the difference toggleable by a parameter for the setup function. => less duplicated code, easier to grasp the difference in the setup.